File: backend/app/schemas/weather.py
from datetime import datetime, date
from typing import List, Optional
from pydantic import BaseModel, Field


class DailyForecast(BaseModel):
    forecast_date: date
    temp_max_c: float
    temp_min_c: float
    humidity_pct: float
    rainfall_prob_pct: float
    expected_rainfall_mm: float
    wind_speed_m_s: float
    et0_mm_day: float
    condition_summary: str  # Sunny, Scattered Showers, Heavy Rain, etc.
    heat_stress_alert: bool = False
    frost_alert: bool = False


class WeatherCurrentResponse(BaseModel):
    latitude: float
    longitude: float
    temperature_c: float
    temp_max_c: float
    temp_min_c: float
    humidity_pct: float
    wind_speed_m_s: float
    rainfall_mm: float
    et0_reference_mm_day: float
    condition: str
    observed_at: datetime
    forecast_7day: List[DailyForecast]


class IrrigationAdvisoryResponse(BaseModel):
    crop_id: str
    growth_stage: str
    current_et0_mm_day: float
    crop_coefficient_kc: float
    crop_evapotranspiration_etc_mm_day: float
    effective_rainfall_mm: float
    net_irrigation_depth_mm: float
    recommended_drip_runtime_hours: float
    irrigation_urgency: str  # Critical, Recommended, Adequate, None (Saturated)
    soil_moisture_depletion_pct: float

File: frontend/src/types/weather.ts
export interface DailyForecast {
  forecast_date: string;
  temp_max_c: number;
  temp_min_c: number;
  humidity_pct: number;
  rainfall_prob_pct: number;
  expected_rainfall_mm: number;
  wind_speed_m_s: number;
  et0_mm_day: number;
  condition_summary: string;
  heat_stress_alert: boolean;
  frost_alert: boolean;
}

export interface WeatherCurrentResponse {
  latitude: number;
  longitude: number;
  temperature_c: number;
  temp_max_c: number;
  temp_min_c: number;
  humidity_pct: number;
  wind_speed_m_s: number;
  rainfall_mm: number;
  et0_reference_mm_day: number;
  condition: string;
  observed_at: string;
  forecast_7day: DailyForecast[];
}

export interface IrrigationAdvisoryResponse {
  crop_id: string;
  growth_stage: string;
  current_et0_mm_day: number;
  crop_coefficient_kc: number;
  crop_evapotranspiration_etc_mm_day: number;
  effective_rainfall_mm: number;
  net_irrigation_depth_mm: number;
  recommended_drip_runtime_hours: number;
  irrigation_urgency: 'Critical' | 'Recommended' | 'Adequate' | 'None';
  soil_moisture_depletion_pct: number;
}
